tinydrive/jwt-authentication.md

```diff
@@ -14,50 +16,27 @@ JWT is a standard authorization solution for web services and is documented in m
-
-The private/public key pair can be created on your [{{site.companyname}} - {{site.accountpage}} page]({{site.accountpageurl}}/jwt/), however we only store the public key on the {{site.accountpage}}. The private key should be downloaded and stored in your backend.
-
-> **Important**: Keep the private key secure, do not commit files containing the key to public repositories or websites.
-
-For information on generating an RSA key pair locally, see: [Creating a private/public key pair for {{site.cloudfilemanager}}](#creatingaprivatepublickeypairfortinydrive).
-
-## JWT provider URL
-
-The easiest way to setup JWT authentication against {{site.cloudname}} services is to create a JWT provider endpoint. This endpoint takes a JSON HTTP POST request and produces a JSON result with the token that the service will then use for all the HTTP requests.
-
-The following diagram explains the JWT call flow:
-
-![JSON Web Token Call Flow]({{site.baseurl}}/images/jwt-call-flow.png "JSON Web Token Call Flow")
+{% include auth/jwt-provider-endpoint-url.md %}
 
 ## JWT requirements
 
-### Algorithm
-
-Our examples use, and we recommend RS256 algorithm. This is a list of supported ones: RS256, RS384, RS512, PS256, PS384, PS512
+{% include auth/jwt-supported-algorithms.md %}
 
 ### Claims
 
 These are like options/data you can send with the JWT token.
 
-* **sub** - _(required)_ Unique string to identify the user. This can be a database ID, hashed email address, or similar identifier.
-* **name** - _(required)_ Full name of the user that will be used for presentation inside {{site.cloudfilemanager}}. When the user uploads a file, this name is presented as the creator of that file.
-* **https://claims.tiny.cloud/drive/root** - (optional) Full path to a {{site.cloudfilemanager}} specific root for example "/johndoe". The user won't be able to see or manage files outside this configured root path.
+<dl>
+<dt><code>sub</code> <em>(required)</em></dt><dd>Unique string to identify the user. This can be a database ID, hashed email address, or similar identifier.</dd>
+<dt><code>name</code> <em>(required)</em></dt><dd>Full name of the user that will be used for presentation inside {{site.cloudfilemanager}}. When the user uploads a file, this name is presented as the creator of that file.</dd>
+<dt><code>https://claims.tiny.cloud/drive/root</code> <em>(optional)</em></dt><dd>Full path to a {{site.cloudfilemanager}} specific root for example "/johndoe". The user won't be able to see or manage files outside this configured root path.</dd>
+</dl>
 
-> **Note**: The "sub" value is a case-sensitive string containing a **String** or **URI** value. The `sub` cannot have a `:` *unless* it is a valid URI or else the callback would fail.
+> **Note**: The "`sub`" value is a case-sensitive string containing a **String** or **URI** value. The `sub` cannot have a `:` *unless* it is a valid URI or else the callback would fail.
```

Comment:
```suggestion
> **Note**: The "`sub`" option is a case-sensitive string containing a **String** or **URI** value. The `sub` option cannot have a `:` *unless* it is a valid URI or else the callback will fail.
```
?

Reply:
From @lnewson's comment elsewhere, this should maybe be `claim`??

Reply:
Agreed, the docs are calling it a `claim` in the JWT auth page, so we should keep the phrasing consistent and use `claim` here as well.

Reply:
updated